server/websocket_test.go

```diff
@@ -0,0 +1,228 @@
+
+		err = p.handleBotWSReconnect("connID", "prevConnID", "originalConnID", channelID)
+		require.NoError(t, err)
+	})
+
+	t.Run("only transcribing job", func(t *testing.T) {
+		state := &channelState{
+			Call: &callState{
+				ID: "callID",
+				Transcription: &jobState{
+					BotConnID: "prevConnID",
+				},
+			},
+		}
+		stateJSON, err := json.Marshal(state)
+		require.NoError(t, err)
+
+		mockDriver.On("RowsNext", "rowsID", mock.AnythingOfType("[]driver.Value")).Return(func(rowsID string, dest []driver.Value) error {
+			require.Equal(t, "rowsID", rowsID)
+			dest[0] = stateJSON
+			return nil
+		}).Once()
+
+		state.Call.Transcription.BotConnID = "connID"
+		expectedStateJSON, err := json.Marshal(state)
+		require.NoError(t, err)
+
+		mockAPI.On("KVSet", "channelID", expectedStateJSON).Return(nil).Once()
+
+		err = p.handleBotWSReconnect("connID", "prevConnID", "originalConnID", channelID)
+		require.NoError(t, err)
+	})
+
+	t.Run("both jobs", func(t *testing.T) {
+		t.Run("recording", func(t *testing.T) {
+			state := &channelState{
+				Call: &callState{
+					ID: "callID",
+					Recording: &jobState{
+						BotConnID: "prevRecordingBotConnID",
+					},
+					Transcription: &jobState{
+						BotConnID: "prevTranscribingBotConnID",
+					},
+				},
+			}
+			stateJSON, err := json.Marshal(state)
+			require.NoError(t, err)
+
+			mockDriver.On("RowsNext", "rowsID", mock.AnythingOfType("[]driver.Value")).Return(func(rowsID string, dest []driver.Value) error {
+				require.Equal(t, "rowsID", rowsID)
+				dest[0] = stateJSON
+				return nil
+			}).Once()
+
+			state.Call.Recording.BotConnID = "newRecordingBotConnID"
+			expectedStateJSON, err := json.Marshal(state)
+			require.NoError(t, err)
+
+			mockAPI.On("KVSet", "channelID", expectedStateJSON).Return(nil).Once()
+
+			err = p.handleBotWSReconnect("newRecordingBotConnID", "prevRecordingBotConnID", "originalConnID", channelID)
+			require.NoError(t, err)
+		})
+
+		t.Run("transcription", func(t *testing.T) {
+			state := &channelState{
+				Call: &callState{
+					ID: "callID",
+					Recording: &jobState{
+						BotConnID: "prevRecordingBotConnID",
+					},
+					Transcription: &jobState{
+						BotConnID: "prevTranscribingBotConnID",
+					},
+				},
+			}
+			stateJSON, err := json.Marshal(state)
+			require.NoError(t, err)
+
+			mockDriver.On("RowsNext", "rowsID", mock.AnythingOfType("[]driver.Value")).Return(func(rowsID string, dest []driver.Value) error {
+				require.Equal(t, "rowsID", rowsID)
+				dest[0] = stateJSON
+				return nil
+			}).Once()
+
+			state.Call.Transcription.BotConnID = "newTranscribingBotConnID"
+			expectedStateJSON, err := json.Marshal(state)
+			require.NoError(t, err)
+
+			mockAPI.On("KVSet", "channelID", expectedStateJSON).Return(nil).Once()
+
+			err = p.handleBotWSReconnect("newTranscribingBotConnID", "prevTranscribingBotConnID", "originalConnID", channelID)
+			require.NoError(t, err)
```

Comment:
Could you explain a bit about what this is testing? It's a little unclear to me looking from the outside.

Reply:
The most important part is the `KVSet` mocked call. We are essentially asserting the state mutation is what's expected given the initial `state` and the `handleBotWSReconnect` method's inputs. Let me add a couple of comments.

Reply:
Done